build_tools/github_actions/configure_ci.py

```diff
@@ -136,6 +136,22 @@ def contains(cls, val):
     ]
 )
 
+NVGPU_PATHS = [
+    # Directly related to NVIDIA GPU code generation and execution:
+    "compiler/src/iree/compiler/Codegen/LLVMGPU/*",
+    "runtime/src/iree/hal/drivers/cuda/*",
+    # Common code likely enough to affect code paths unique to NVIDIA GPUs:
+    "compiler/src/iree/compiler/GlobalOptimization/*",
+]
+
+AMDGPU_PATHS = [
+    # Directly related to AMDGPU code generation and execution:
+    "compiler/src/iree/compiler/Codegen/LLVMGPU/*",
+    "runtime/src/iree/hal/drivers/hip/*",
+    # Common code likely enough to affect code paths unique to AMDGPU:
+    "compiler/src/iree/compiler/GlobalOptimization/*",
+]
+
 # Jobs to run in presumbit if files under the corresponding path see changes.
```

Comment:
> I also want to change `build_test_runtime :: arm64` from `ci.yml` and `Regression Test / test_onnx :: amdgpu_vulkan` from `pkgci_regression_test.yml`, but those jobs are under matrices that can't use
> 
> ```yaml
> if: contains(fromJson(needs.setup.outputs.enabled-jobs), 'job_name')
> ```
> 
> At least the pkgci jobs are not blocking PR merges right now.

Definitely want `build_test_runtime :: arm64` excluded too. That competes with `build_test_all_arm64` for a single runner when LLVM integrate PRs are running 1h+ builds like https://github.com/iree-org/iree/actions/runs/9960816403/job/27520932492.

Strategies: https://stackoverflow.com/questions/65384420/how-do-i-make-a-github-action-matrix-element-conditional